inductiva/fluids/scenarios/wind_terrain/wind_terrain.py

```diff
@@ -0,0 +1,205 @@
+    through a mesh. The wind is injected through one of the side walls with
+    a certain vectorail velocity and position. Here, the wind is initialized
+    only on a circular region of the wall to simulate gusts of wind.
+
+   Schematic representation of the Wind and Simulation scenario:
+
+         Wind Injection                 2D Representation of Simulation
+     ____________________                     ____________________
+    |                    |                   |                    |
+    |    * *             |                   |                    |
+    |  *     *           |                   |->                  |
+    |  *     *           |                   |->          /\      |
+    |    * *             |                   |         /\/  \     |
+    |____________________|                   |________/______\____|
+
+    This scenario solves the steady-state continuity and momentum equations
+    (time-independent) with the assumption of incompressible flow. 
+    The simulation solves the time-independent equations for several
+    time steps, based on the state of the previous one. The end goal is
+    to determine the steady-state of the system, i.e., where the flow
+    does not change in time anymore.
+
+    This scenarion can be simulation with OpenFOAM.
+    """
+
+    valid_simulators = [fluids.simulators.OpenFOAM]
+
+    def __init__(self,
+                 terrain: inductiva.world.Terrain,
+                 wind_velocity: List[float],
+                 wind_position: Optional[List[float]] = None,
+                 athmosphere_height: float = 300):
+        """Initializes the `WindTerrain` conditions.
+
+        Args:
+            wind_velocity (List): Velocity of the air flow (m/s).
+            wind_position (List): Position of the wind flow (m) above the height
+            of the terrain.
+            terrain (inductiva.world.Terrain): Terrain object that describes
+                the profile of the terrain with a mesh.
+            athmosphere_height (float): Altitude above the terrain (m) to establish an
+                atmosphere. Notice that the wind_position needs to be inside the
+                atmosphere region.
+        """
+
+        self.wind_velocity = np.array(wind_velocity)
+        # Compute the wind_direction from the velocity vector.
+        self.wind_direction = self.wind_velocity / np.linalg.norm(
+            self.wind_velocity)
+        if wind_position is None:
+            wind_position = [
+                terrain.center["x"], terrain.center["y"],
+                terrain.bounds["z"][1] + 10
+            ]
+
+        self.wind_position = wind_position
+        self.terrain = terrain
+
+        self.top_boundary_height = terrain.bounds["z"][0] + athmosphere_height
+
+        if terrain.bounds["z"][1] >= self.top_boundary_height:
+            raise ValueError(
+                "The terrain height surpasses the top athmosphere boundary.")
```

Comment:
Usually you would check the validity of the inputs before assigning them to class attributes, instead of performing validity checks on `self....`